### PR TITLE
[FlagGems Operator Development Competition] smooth_l1_loss

### PR DESCRIPTION
**Pull request by shido29200**
## FlagGems Operator Development Competition

**Operator:** `smooth_l1_loss` / `smooth_l1_loss_backward` (aten::smooth_l1_loss, aten::smooth_l1_loss_backward)
**Category:** Medium

## Implementation

Implements forward and backward passes for Smooth L1 Loss (Huber Loss) using Triton kernels.

### Formula

```
L2 region (|x−y| < β):  loss = 0.5 * (x−y)² / β
L1 region (|x−y| ≥ β):  loss = |x−y| − 0.5 * β
```

### Architecture

**Forward (`reduction='none'`):**  
`@pointwise_dynamic` with `beta` as a scalar argument (`is_tensor=[True, True, False]`).  
Fully branchless via `tl.where`, coalesced memory access, zero kernel-launch overhead.

**Forward (`reduction='mean'/'sum'`):**  
2-pass kernel matching the `mse_loss` pattern:
1. `smooth_l1_fwd_kernel` — per-block partial Huber sum
2. `smooth_l1_reduce_kernel` — final scalar accumulation

**Backward (all reductions):**  
Single `@pointwise_dynamic` kernel; `scale = 1/N` for `mean`, `1.0` otherwise.

```
L2 gradient: diff / beta
L1 gradient: sign(diff)
```

## Files changed

- `src/flag_gems/ops/smooth_l1_loss.py` — Triton kernels + Python wrappers
- `src/flag_gems/ops/__init__.py` — export `smooth_l1_loss`, `smooth_l1_loss_backward`
- `src/flag_gems/__init__.py` — register `aten::smooth_l1_loss` and `aten::smooth_l1_loss_backward`
- `tests/test_reduction_ops.py` — accuracy tests (FLOAT_DTYPES × REDUCTION_SHAPES × reductions × betas)
- `benchmark/test_reduction_perf.py` — performance benchmark alongside `mse_loss`

## Test plan

- [ ] Accuracy forward: `pytest tests/test_reduction_ops.py -k smooth_l1_loss`
- [ ] Accuracy backward: `pytest tests/test_reduction_ops.py -k smooth_l1_loss_backward`
- [ ] Performance: `pytest benchmark/test_reduction_perf.py -k smooth_l1_loss`

### CODE REVIEW

**Comment by douxetpur** [comment]
Hi, thank you for your participation in the FlagOS Open Computing Competition 🙏

After review, we have chosen to move forward with a different implementation for this operator. As a result, this PR will be closed.

We appreciate your time and contribution, and hope to see more of your submissions in the future.
